### ISSUE TITLE
Set up test bed for Peter to test our parallel, future, etc

### ISSUE THREAD

**Issue by ColSmith76** [RSGInc/cmap_freight_model#1]
Need a model test bed for Peter to work in -- small example version of model to run limited set of markets and evaluate performance of parallel, future, etc.

**Comment by ColSmith76** [comment]
I created a new branch in the repository for you which now runs without any additional intervention (just added a line to make sure there is a ./library folder created if it is not already present). The branch is called “parallel” and I also cloned a local copy on the WRJMDLPPW01 machine, E:\Projects\Clients\CMAP\cmap_freight_parallel

Some details:

1.	./scripts/0_Main.R --- this is the main script that controls model flow.
a.	Lines 62-65 runs the 4 model components that we are interested in for this project. Everything from step 5 onwards is a connect model of just truck movements in Chicago that is not relevant for this project. The output at the end of step 4 (pairs.Rdata from line 160 of 04_PMG_Outputs.R) is what we are getting to).
b.	The first step (01_Firm_Synthesis.R) is firm synthesis which creates a set of separate industry/commodity markets that we will simulate. The naics_set created on line 504 of 01_Firm_Synthesis.R describes the markets and how they are to be divided up for simulation into groups. (NAICS is the North American Industrial Classification System). 
c.	The second step (02_Procurement_Markets.R) prepares a work space for the separate processes to use during simulation
d.	The third step (03_PMG_Controller.R) runs two loops that use a set of spawned off R script processes. The first creates a set of inputs files for each individual procurement market simulation. The second runs the individual procurement market simulations (which is the C++ executable, E:\Projects\Clients\CMAP\cmap_freight_parallel\PMG\PMG.exe
e.	The fourth step (04_PMG_Outputs.R) gathers together and processes the outputs from all of the individual procurement market simulations to create the pairs table, which represents the successful trades for all of the commodity markets.
2.	The issue that the client is having is that at line 37 and line 85 of 03_PMG_Controller.R calls tasklist.exe and this sometimes crashes. You’ll see that thew approach to running the parallel process is something like this, run within the while loops at lines 34 and 71:
a.	See how many processes are running
b.	If its less than the max allowed start another
c.	Test again after 30 seconds
d.	Keep going until all of the naicstorun are completed
3.	I’d suggest replaceing the use of tasklist with either something from future or the parallel package, e.g. clusterApplyLB(). 
a.	The two scripts that need to be run in parallel are 03_0a_Supplier_to_Buyer_Costs.R called on line 51 and 03a_Run_PMG.R called on line 99.
b.	The approach that is currently implemented – to run the naics markets in parallel, and then (within the separate parallel processes) to run the groups in sequence is not really necessary. They groups are independent and in theory we could create a single large set of naics-groups combinations and runs those all in parallel. That would require some reconfiguation of what happens in 03_0a_Supplier_to_Buyer_Costs.R and 03a_Run_PMG.R but would likely be preferable as the numbers of groups in a market varies tremendously and some markets can take a long time to run, meaning that other work has completed and one or two of the parallel processes are left running for a long time.
